simulation/adx_gyro_accelerometer/src/main/native/cpp/ADXRS450_SpiGyroWrapperData.cpp

```diff
@@ -38,6 +63,48 @@ void ADXRS450_SpiGyroWrapper::HandleRead(uint8_t* buffer, uint32_t count) {
   std::memcpy(&buffer[0], &returnCode, sizeof(returnCode));
 }
 
+void ADXRS450_SpiGyroWrapper::HandleAutoReceiveData(uint8_t* buffer,
+                                                    int32_t numToRead,
+                                                    int32_t& outputCount) {
+  double diff = m_angleDiff;
+  int32_t messagesToSend =
+      std::abs(diff > 0 ? std::ceil(diff / kMaxAngleDeltaPerMessage)
+                        : std::floor(diff / kMaxAngleDeltaPerMessage));
+
+  // Zero gets passed in during the "How much data do I need to read" step.
+  // Else it is actually reading the accumulator
+  if (numToRead == 0) {
+    outputCount = messagesToSend * kPacketSize;
+    return;
+  }
+
+  int valuesToRead = numToRead / kPacketSize;
+  std::memset(&buffer[0], 0, numToRead);
+
+  int msgCtr = 0;
+
+  while (msgCtr < valuesToRead) {
+    double cappedDiff = diff;
+    if (cappedDiff > kMaxAngleDeltaPerMessage) {
```

Comment:
You could use the clamp() implementation from PIDController.cpp here for clarity. I don't know where would be best to put the clamp() implementation though. If it were for wpilibc only, I'd put it in wpilibc's Base.h because that's where we put C++ standard library shims (std::clamp() is a C++17 feature).

Reply:
This doesn't know about wpilibc.  I changed it to do a min/max and put a comment in for clarity